query-translator/tests/translate.rs

```diff
@@ -286,6 +286,50 @@ fn test_unknown_ident() {
     assert_eq!("SELECT 1 LIMIT 0", sql);
 }
 
+#[test]
+fn test_type_required_long() {
+    let schema = Schema::default();
+
+    let query = r#"[:find ?x :where [?x _ ?e] [(long ?e)]]"#;
+    let SQLQuery { sql, args } = translate(&schema, query);
+
+    assert_eq!(sql, "SELECT DISTINCT `all_datoms00`.e AS `?x` \
+                     FROM `all_datoms` AS `all_datoms00` \
+                     WHERE (`all_datoms00`.value_type_tag = 5 AND \
+                            typeof(`all_datoms00`.v) = 'integer')");
+
+    assert_eq!(args, vec![]);
+}
+
+#[test]
+fn test_type_required_double() {
+    let schema = Schema::default();
+
+    let query = r#"[:find ?x :where [?x _ ?e] [(double ?e)]]"#;
```

Comment:
Here's a question for you. If we rephrase this as

`[:find ?x :where [(double ?e)] [?x _ ?e]]`

do we still use `all_datoms` rather than `datoms`? If so, please file a bug for that.

Reply:
Just so that it's recorded, this is no longer an issue as of b0d96e598a6f29e6dd8701279860e423d895ae72 (commit 4), which ensures that we use top level type hints when determining which table to use, even if they appear late.